nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java

```diff
@@ -271,4 +302,59 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         }
 
     }
+
+    @OnScheduled
+    public void createClient(ProcessContext context) throws IOException {
```

Comment:
Most of this method appears to duplicate `AbstractMongoProcessor#createClient()`. I suggest the duplicated code should be refactored.

Reply:
It is true I could have done the changes in AbstractMongoProcessor, but it would end up affecting the components of PutMongo, if the ontrigger method of it's not modified, there would be an issue. Also Since request was for just getMongo I did it this way. 


Regardless, do we have to create a new ticket to change processors getMongo and Delete Mongo or can I do with this itself?

Reply:
I think all 6 of the MongoDB processors need this change if one does, so it should be made in a consistent manner. It appears the Jira is incomplete as it only indicates the `GetMongo` processor, but that's what this review process is intended to mitigate. Requirements gathering is notoriously difficult and a single user's request has to be balanced against the ongoing needs of the project as a whole. Any code changes introduced need to be well-understood, complete, tested, sustainable, and maintainable. I use the analogy from camping of "leave it better than you found it."

Reply:
@alopresto We started moving toward using a controller service to manage the client connection I think ~3 releases ago. The best place for these changes might be there, so we can start moving toward deprecating per-processor connection management altogether.